sycl/test/kernel-and-program/kernel-and-program.cpp

```diff
@@ -101,6 +99,28 @@ int main() {
         assert(a == b + c);
       }
     }
+    {
+      cl::sycl::queue Queue;
+      if (!Queue.is_host()) {
+        cl::sycl::sampler first(
+            cl::sycl::coordinate_normalization_mode::normalized,
+            cl::sycl::addressing_mode::clamp, cl::sycl::filtering_mode::linear);
+        cl::sycl::sampler second(
+            cl::sycl::coordinate_normalization_mode::unnormalized,
+            cl::sycl::addressing_mode::clamp_to_edge,
+            cl::sycl::filtering_mode::nearest);
+        cl::sycl::program prog(Queue.get_context());
+        prog.build_with_source(
+            "kernel void sampler_args(int a, sampler_t first, "
+            "int b, sampler_t second, int c) {}\n");
```

Comment:
I'm not sure I understand what we test here. 
Shouldn't we check that sampler is passed to the kernel is what we initialized at lines ##105-111?

Reply:
Here we check that the SYCL sampler object can be passed as argument  for OpenCL kernel that was built from source.
Otherwise we'd get an error like CL_INVALID_SAMPLER. 
Do you think that we should check the passed samplers inside the kernel also?

Reply:
It would be great, although I'm not sure how it can be done w/o image support.
Any ideas?

Reply:
I can try to use OpenCL API to create and to pass OpenCL image object into this kernel and do something to check if samplers were passed to the kernel correctly.

Reply:
Let's do this when image class will be supported. When it will be happened then will be possible to write an integration test that will use class sampler and class image together.